internal/fs/inode/dir.go

```diff
@@ -1034,6 +1037,93 @@ func (d *dirInode) DeleteChildDir(
 	return nil
 }
 
+// LOCKS_REQUIRED(d)
+func (d *dirInode) DeleteObjects(ctx context.Context, objectNames []string) error {
+	for _, objectName := range objectNames {
+		if strings.HasSuffix(objectName, "/") {
+			// Initiate deletion for a prefix (directory). This logic handles pagination internally.
+			if err := d.deletePrefixRecursively(ctx, objectName); err != nil {
+				return fmt.Errorf("recursively deleting prefix %q: %w", objectName, err)
+			}
+		} else {
+			// Handle single file-like object deletion.
+			if err := d.deleteSingleObject(ctx, objectName); err != nil {
+				return fmt.Errorf("deleting unsupported object %q: %w", objectName, err)
+			}
+		}
+	}
+	return nil
+}
+
+// Helper to delete a single object, handling 'Not Found' errors gracefully.
+func (d *dirInode) deleteSingleObject(ctx context.Context, objectName string) error {
+	if d.isBucketHierarchical() && strings.HasSuffix(objectName, "/") {
+		if err := d.bucket.DeleteFolder(ctx, objectName); err != nil {
+			var notFoundErr *gcs.NotFoundError
+			if !errors.As(err, &notFoundErr) {
+				return err
+			}
+		}
+	} else {
+		if err := d.bucket.DeleteObject(ctx, &gcs.DeleteObjectRequest{Name: objectName}); err != nil {
+			var notFoundErr *gcs.NotFoundError
+			if !errors.As(err, &notFoundErr) {
+				return err
+			}
```

Comment:
what about typecache - dont we need to delete type cache entries?

Reply:
We have confirmed that unsupported directories correctly excluded from type cache insertion during the following operations:
1. [List](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L760)
2. Copy - Kernel can not read unsupported objects so it won't be able to send copy object.
3. [Rename](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/inode/dir.go#L1076)

This exclusion eliminates the need for any subsequent cleanup logic, meaning we no longer require a specific remove operation to delete these unsupported entries from the type cache.